components/loki.libsonnet

```diff
@@ -0,0 +1,624 @@
+            timeout: '100ms',
+            max_idle_conns: 100,
+            update_interval: '1m',
+            consistent_hash: true,
+          },
+        },
+      },
+    },
+  },
+
+  withIndexQueryCache:: {
+    local l = self,
+    config+:: {
+      indexQueryCache: error 'must provide addresses for index query cache',
+    },
+
+    defaultConfig+:: {
+      storage_config+: {
+        index_queries_cache_config: {
+          memcached: {
+            batch_size: 100,
+            parallelism: 100,
+          },
+          memcached_client: {
+            addresses: l.config.indexQueryCache,
+            consistent_hash: true,
+          },
+        },
+      },
+    },
+  },
+
+  withIndexWriteCache:: {
+    local l = self,
+    config+:: {
+      indexWriteCache: error 'must provide addresses for index writes cache',
+    },
+
+    defaultConfig+:: {
+      chunk_store_config+: {
+        write_dedupe_cache_config: {
+          memcached: {
+            batch_size: 100,
+            parallelism: 100,
+          },
+          memcached_client: {
+            addresses: l.config.indexWriteCache,
+            consistent_hash: true,
+          },
+        },
+      },
+    },
+  },
+
+  withResultsCache:: {
+    local l = self,
+    config+:: {
+      resultsCache: error 'must provide addresses for frontend results cache',
+    },
+
+    defaultConfig+:: {
+      query_range+: {
+        split_queries_by_interval: '30m',
+        align_queries_with_step: true,
+        cache_results: true,
+        max_retries: 5,
+        results_cache: {
+          max_freshness: '10m',
+          cache: {
+            memcached_client: {
+              timeout: '500ms',
+              consistent_hash: true,
+              addresses: l.config.resultsCache,
+              update_interval: '1m',
+              max_idle_conns: 16,
+            },
+          },
+        },
+      },
+    },
+  },
+
+  withEtcd:: {
+    local l = self,
+
+    config+:: {
+      etcdEndpoints: error 'must provide etcd endpoints list',
+    },
+
+    defaultConfig+:: {
+      distributor+: {
+        ring: {
+          kvstore: {
+            store: 'etcd',
+            etcd: {
+              endpoints: l.config.etcdEndpoints,
+            },
+          },
+        },
+      },
+      ingester+: {
+        lifecycler+: {
+          ring+: {
+            kvstore: {
+              store: 'etcd',
+              etcd: {
+                endpoints: l.config.etcdEndpoints,
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+
+  withMemberList:: {
+    local l = self,
+    local gossipRingName = 'gossip-ring',
+    local gossipPort = 7946,
+    local gossipSvc =
+      local service = k.core.v1.service;
+      local ports = service.mixin.spec.portsType;
+      local name = l.config.name + '-' + gossipRingName;
+
+      service.new(
+        name,
+        l.config.podLabelSelector { 'app.kubernetes.io/gossip': 'true' },
```

Comment:
we should namespace this label to something other than `app.kubernetes.io` since it's not a well-known or recommended label from k8s: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels. consider `loki.grafana.com/gossip`?

Reply:
I addressed this point. You are absolutely right!